airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/state/CheckpointManager.kt

```diff
@@ -155,7 +164,13 @@ abstract class StreamsCheckpointManager<T> : CheckpointManager<DestinationStream
             val head = globalCheckpoints.peek()
             val allStreamsPersisted =
                 head.streamIndexes.all { (stream, index) ->
-                    syncManager.getStreamManager(stream).areRecordsPersistedUntil(index)
+                    if (!checkpointById) {
```

Comment:
nit: invert this if statement (i.e. `if (checkpointById) { ... CheckpointId()... } else {...}`)